src/dlio/odom.cc

```diff
@@ -447,6 +449,53 @@ void dlio::OdomNode::publishToROS(pcl::PointCloud<PointType>::ConstPtr published
   br.sendTransform(transformStamped);
 }
 
+// Function for correcting the pointcloud tilestamp to more accurate represent its pose
+ros::Time dlio::OdomNode::correctTimestamp(ros::Time original_stamp) {
+  
+  // Pose buffer is empty, return previous stamp
+  if (this->pose_buffer.empty()) {
+    printf("Correct timestamp failed 1\n");
+    return original_stamp;
+  }
+
+  // Required variables
+  auto curr_pose = this->pose_buffer.begin(); // Current pose
+  auto prev_pose = this->pose_buffer.begin(); // Previous pose
```

Comment:
No poses are ever popped from the buffer, this will currently grow till `this->imu_buffer_size_`, which is currently 5000